src/main/proto/marathon.proto

```diff
@@ -132,7 +132,7 @@ message ServiceDefinition {
   optional int64 last_scaling_at = 23;
   optional int64 last_config_change_at = 24;
   optional ObsoleteIpAddress OBSOLETE_ipAddress = 25; // deprecated since 1.5
-  optional ResidencyDefinition residency = 26;
+  optional ResidencyDefinition residency = 26 [deprecated = true];
```

Comment:
I remember that the way fields of `marathon.proto` are way different from the approach Protobuf docs suggest. @jdef is probably the one to talk to. Basically, it is about avoiding `[deprecated = true]`, and renaming fields instead (something like `residency -> DEPRECATED_residency`). @timcharper it would be nice to have your take on this.